crates/load-cargo/src/lib.rs

```diff
@@ -413,4 +443,203 @@ mod tests {
         // RA has quite a few crates, but the exact count doesn't matter
         assert!(n_crates > 20);
     }
+
+    mod source_root_parent {
+        use ide_db::base_db::SourceRootId;
+        use vfs::{file_set::FileSetConfigBuilder, VfsPath};
+
+        use crate::SourceRootConfig;
+
+        macro_rules! virp {
+            ($s : literal) => {
+                VfsPath::new_virtual_path(format!($s))
+            };
+        }
+
+        #[test]
+        fn test1() {
```

Comment:
What do we actually think about enumerating tests under a module that groups all tests with similar purpose?

Reply:
Makes it tough to reason about what each test is testing

Reply:
not really, i'd just name each test after what it does and move it under the `test` module above—it's not as if this is gated behind a `cfg`.